### PR TITLE
Add Section for Module Collections

### LINKED ISSUE #669: Beware of `module save` becoming the user's modules-loaded-by-default
### What would you like to see added?

See here for instructions on how this can happen <https://lmod.readthedocs.io/en/latest/010_user.html#user-collections>.

If a researcher uses `module save $name` at any point, the currently loaded set of modules is saved as a user collection in a configuration file used by lmod, with the name `$name`. Collections can be loaded with `module restore $name`, and listed with `module savelist`.

If the command `module save` is given with no `$name`, then it is saved with the name `default` and loaded as part of the lmod startup process. Every shell and terminal will now have a command equivalent to `module restore default` run, restoring those saved default modules. This is a new way for researchers to have issues starting OOD jobs, including HPC Desktop jobs, if the modules loaded include Anaconda or other modules that may interfere with OOD jobs functioning.

To disable any module collection, including the default collection, use `module disable $name` or `module disable default`. To delete a collection, navigate to `~/lmod.d/` and delete the file named `$name`. If it had previously been disabled, it will have a `~` character at the end like `$name~`.

### PR DESCRIPTION
**Pull request by Premas**
# Pull Request

## Overview
Added a section for Module Collections with best practices to prevent VNC errors and related issues.

## Proposed Changes
Added a section for Module Collections

## Related Issues
Fixes #669

### CODE REVIEW

File: docs/cheaha/software/modules.md [hunk 1]
```diff
@@ -52,6 +52,42 @@ If you want to revert to the default modules, you can use:
 module reset
 ```
 
+## Saving Modules using Collections
+
+To save time in typing in long list of modules everytime you work on a project, you can save the desired list of modules using module collection. To acheive this first load the desired modules, and save it using a collection name as below,
```
Comment (wwarriner) [inline docs/cheaha/software/modules.md:57]: At the end of the first sentence, please change to "using a module collection."

I avoid using "ordering" words in writing, as they add clutter but not much clarity. So, in the second sentence, "To achieve this, load the desired modules and save them to a collection, as shown below." This is a style choice, not required, but I think it flows well.

File: docs/cheaha/software/modules.md [hunk 2]
```diff
@@ -52,6 +52,42 @@ If you want to revert to the default modules, you can use:
...
+
+```bash
+module load module 1 module 2 ...
```
Comment (wwarriner) [inline docs/cheaha/software/modules.md:60]: `module_1` and `module_2` with underscores will help with clarity.

**Review by wwarriner — CHANGES_REQUESTED** [review]
Looks great! A couple of minor changes.

File: docs/cheaha/software/modules.md (hunk 2 above)
Comment (Premas) [reply docs/cheaha/software/modules.md]: Added underscores.

File: docs/cheaha/software/modules.md (hunk 1 above)
Comment (Premas) [reply docs/cheaha/software/modules.md]: Sure. Incorporated the changes.

**Comment by Premas** [comment]
Thanks for reviewing!

**Comment by wwarriner** [comment]
Everything looks good, I'm going to integrate this one.

**Pull request merged**